src/pcsx2_ipc.h

```diff
@@ -815,6 +829,46 @@ class PCSX2Ipc {
         return StringCommands<tag>();
     }
 
+    /**
+     * Retrieves emulator status. @n
+     * On error throws an IPCStatus. @n
+     * Format: XX @n
+     * Legend: XX = IPC Tag. @n
+     * Return: ZZ ZZ ZZ ZZ @n
+     * Legend: ZZ = emulator status.
+     * @see IPCCommand
+     * @see IPCStatus
+     * @param T Flag to enable batch processing or not.
+     * @return If in batch mode the IPC message otherwise the emulator status.
+     */
+    template <bool T = false>
+    auto Status() {
+        constexpr IPCCommand tag = MsgStatus;
+        // batch mode
+        if constexpr (T) {
+            if (BatchSafetyChecks(1, 4)) {
+                SetError(OutOfMemory);
+                return (char*)0;
+            }
+            char* cmd = &ipc_buffer[batch_len];
+            cmd[0] = tag;
+            batch_len += 1;
+            batch_arg_place[arg_cnt] = reply_len;
+            reply_len += 4;
+            arg_cnt += 1;
+            return cmd;
+        }
+        else {
+            // we are already locked in batch mode
+            std::lock_guard<std::mutex> lock(ipc_blocking);
+            ToArray(ipc_buffer, 4 + 1, 0);
+            ipc_buffer[4] = tag;
+            SendCommand(IPCBuffer{ 4 + 1, ipc_buffer },
+                        IPCBuffer{ 1 + 4 + 4, ret_buffer });
```

Comment:
please keep the order consistent with the packet tags so we can see the order of data, 4 + 1 + 4 here

Reply:
fixed